kerastuner/engine/metrics_tracking.py

```diff
@@ -22,136 +22,220 @@
 from tensorflow import keras
 
 
-MetricObservation = collections.namedtuple(
-    'MetricObservation',
-    'value step')
+class Direction(object):
```

Comment:
I'm skeptical we should have a class that wraps a string plus a sorting function. `MetricTracker` is the most reasonable place where to centralize metric sorting.

1) It's not fun to hit things like `obj.direction` being a `Direction` object instead of something you can inspect like a string (this was a common source of issue with e.g. `TensorShape` which wrapped a bunch of `Dimension` objects instead of single ints / None).
2) This abstraction does not meaningfully reduce code quantity, it slightly increases it
3) Now readers of code that calls a `Direction` have one more place to check to follow what's being executed, vs. in-lining a sorting function in the 2 places that need it

The Zen of Python: *Simple is better than complex* / *Flat is better than nested.*

Reply:
Done. Makes sense, yeah I was thinking `Direction` would reduce complexity but it didn't end up being needed much, removed it